lib/components/molecules/Table/Table.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+
+import React, { Children } from 'react';
+import styled from 'styled-components';
+
+import styles from './Table.style';
+import type { Props } from './types';
+
+const Table = ({ children, data, className, vertical, ...props }: Props) => {
+  const Head = child => (
+    <th key={child.props.name} role="columnheader">
+      {child.props.children}
```

Comment:
There is no real need to keep the Head under Table. It can just be a small functional component in same file but outside and then you will not need to do this {child.props.children} instead of {children}